mass_mailing_event/models/event_registration.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, api
+from .mail_mass_mailing import event_filtered_ids
+
+
+class EventRegistration(models.Model):
+    _inherit = 'event.registration'
+
+    @api.model
+    def search_count(self, domain):
```

Comment:
I don't need to overwrite ```search```. There is only two actions to hook:

1. search_count: In order to show in the form the right number of recipients.
2. get_recipients: In order to exclude the ones that are registered in event.

```get_recipients``` calls to ```search``` internally but I've preferred to overwrite ```get_recipients``` because there is only one and it is more specific.